### PR TITLE
chore(deps): upgrade to rsmt2d v0.12.0

### PR DESCRIPTION
**Pull request by rootulp**
Bump to the latest rsmt2d release candidate. If CI passes, I'll cut an official v0.12.0 release of rsmt2d and bump to that here.

### CODE REVIEW

**Comment by rootulp** [comment]
The docker-build CI tasks are expected to fail until we merge https://github.com/celestiaorg/celestia-app/pull/3085 so rsmt2d looks good to me.

**Comment by rootulp** [comment]
Update: cut an official release and bumped here.

**Review by ninabarbakadze — APPROVED** [review]

File: test/testground/go.mod [hunk 1]
```diff
@@ -11,12 +11,12 @@ require (
 )
 
 require (
-	cloud.google.com/go v0.110.8 // indirect
```
Comment (staheri14) [inline test/testground/go.mod:14]: [Question] Curious to know why the testground's `go.mod` underwent these changes? were they necessary?

Comment (staheri14) [reply test/testground/go.mod:14]: Never mind, these are indirect

**Review by staheri14 — APPROVED** [review]

**Review by staheri14 — APPROVED** [review]

**Review by cmwaters — APPROVED** [review]

**Pull request merged**